examples/next-openai/app/stream-react-response/action.tsx

```diff
@@ -0,0 +1,166 @@
+'use server';
+
+import {
+  JSONValue,
+  Message,
+  OpenAIStream,
+  experimental_StreamData,
+  experimental_StreamingReactResponse,
+} from 'ai';
+import OpenAI from 'openai';
+import { ChatCompletionCreateParams } from 'openai/resources/chat';
+
+const openai = new OpenAI({
+  apiKey: process.env.OPENAI_API_KEY!,
+});
+
+const functions: ChatCompletionCreateParams.Function[] = [
+  {
+    name: 'get_current_weather',
+    description: 'Get the current weather',
+    parameters: {
+      type: 'object',
+      properties: {
+        location: {
+          type: 'string',
+          description: 'The city and state, e.g. San Francisco, CA',
+        },
+        format: {
+          type: 'string',
+          enum: ['celsius', 'fahrenheit'],
+          description:
+            'The temperature unit to use. Infer this from the users location.',
+        },
+      },
+      required: ['location', 'format'],
+    },
+  },
+  {
+    name: 'create_image',
+    description: 'Create an image for the given description',
+    parameters: {
+      type: 'object',
+      properties: {
+        description: {
+          type: 'string',
+          description: 'Description of what the image should be.',
+        },
+      },
+      required: ['description'],
+    },
+  },
+];
+
+export async function handler({ messages }: { messages: Message[] }) {
+  const data = new experimental_StreamData();
+
+  // Request the OpenAI API for the response based on the prompt
+  const response = await openai.chat.completions.create({
+    model: 'gpt-3.5-turbo',
+    stream: true,
+    messages: messages.map(
+      m =>
+        ({
+          role: m.role,
+          content: m.content,
+        } as any), // TODO cleanup
```

Comment:
The is a bug lurking here, bc OpenAI made their types way more restrictive and our message format does not match any more. We might need to either change it or write an adapter function.